ethereumV2/contracts/factory/Factory.sol

```diff
@@ -155,6 +185,11 @@ contract Factory is OwnableContract {
 
         (nonce, request) = getPendingMintRequest(requestHash);
 
+        if (feed != address(0)) {
+            string memory err = validateProofOfReserves(request.amount);
+            require(bytes(err).length == 0, err);
```

Comment:
Why do this check instead of just raising the error in `validateProofOfReserves`? Looks like the method is only used in one place.